Server/Services/LeaderBoardService.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Server.Controllers;
+using Server.Models;
+
+namespace Server.Services
+{
+    public class LeaderBoardService
+    {
+        private readonly IConfiguration _configuration;
+        private readonly AppDbContext _context;
+        private readonly int _numberOfTopPlayers;
+        private readonly int _maximumDelta;
+        private readonly int _baseRating;
+        private readonly int _divisionPoints;
+
+        public LeaderBoardService(IConfiguration configuration, AppDbContext context)
+        {
+            _configuration = configuration;
+            _context = context;
+            _numberOfTopPlayers = Int32.Parse(configuration["NumberTopPlayers"]);
+            _maximumDelta = Int32.Parse(configuration["MaximumDelta"]);
+            _baseRating = Int32.Parse(configuration["BaseRating"]);
+            _divisionPoints = Int32.Parse(configuration["DivisionPoints"]);
+        }
+
+        /// <summary>
+        /// Should return the top 100 top players. If you need to change that, change the constant at the beginning
+        /// </summary>
+        /// <returns></returns>
+        public async Task<List<LeaderBoardPlayer>> GetTopPlayers()
+        {
+            return await _context.Players
+                            .OrderByDescending(p => p.Rating12v12)
+                            .Take(_numberOfTopPlayers)
+                            .Select(p => new LeaderBoardPlayer { SteamId = p.SteamId, Rating = p.Rating12v12 })
+                            .ToListAsync();
+        }
+
+        /// <summary>
+        /// It gets winning teams, losing teams and then updates the new rating
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="players"></param>
+        /// <returns></returns>
+        public async Task<List<Player>> UpdateNewRating(AfterMatchRequest request, List<Player> players)
+        {
+            var winningTeam = GetWinningTeam(request, players, out var averageWinningRating);
+            var losingTeam = GetLosingTeam(request, players, out var averageLosingRating);
+            var scoreDelta = CalculateScoreDelta(averageWinningRating, averageLosingRating);
+            await UpdateRating(winningTeam, losingTeam, scoreDelta);
+            return players;
+        }
+
+
+        /// <summary>
+        /// Formula is the difference between loosing and winning team of avg rating divided by 40
```

Comment:
That sounds more like an implementation detail, in which case I'd just move it to the method body.